packages/radix-ui-themes/src/components/dialog.css

```diff
@@ -1,46 +1,43 @@
 .rt-DialogOverlay {
-  display: flex;
-  align-items: center;
-  justify-content: center;
   position: fixed;
   inset: 0;
 
-  --dialog-overlay-padding-top: var(--space-4);
-  --dialog-overlay-padding-bottom: max(var(--space-4), 4vh);
-  padding-top: var(--dialog-overlay-padding-top);
-  padding-bottom: var(--dialog-overlay-padding-bottom);
-  padding-left: var(--space-4);
-  padding-right: var(--space-4);
-
-  &::after {
+  &::before {
+    position: fixed;
     content: '';
-    position: absolute;
     inset: 0;
     background-color: var(--color-overlay);
   }
 }
 
+.rt-DialogScroll {
+  display: flex;
+  overflow: auto;
+  position: absolute;
+  inset: 0;
+}
+
+.rt-DialogScrollPadding {
+  margin: auto;
+  padding-top: var(--space-6);
+  padding-bottom: max(var(--space-6), 6vh);
+  padding-left: var(--space-4);
+  padding-right: var(--space-4);
+}
```

Comment:
An extra element is needed because padding-bottom doesn't add any bottom padding in overflowing scroll containers